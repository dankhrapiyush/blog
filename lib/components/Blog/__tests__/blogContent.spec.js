import React from 'react';
import TestRenderer from 'react-test-renderer';
import BlogContent from '../blogContent';
import { IntlProvider } from 'react-intl';

jest.mock('@magento/venia-ui/lib/classify');
jest.mock('react-router-dom', () => ({
    Link: ({ children, ...rest }) => <div {...rest}>{children}</div>,
    html: x => x
}));
jest.mock('@magento/peregrine/lib/util/makeUrl');
jest.mock('@magento/venia-ui/lib/components/RichContent', () => 'RichContent');

const blogContentProps = {
    data: {
        post_id: 1,
        title: 'Foo Bar',
        content: 'Foo Bar Description',
        publish_time: '2020-07-10 00:00:00',
        author: {
            title: 'web dev',
            name: 'web dev',
            url: null,
            author_url: 'https://magento.local/blog/author/web-dev'
        },
        tags: [
            {
                tag_id: 1,
                identifier: 'tag1',
                title: 'tag1',
                is_active: 1
            },
            {
                tag_id: 2,
                identifier: 'tag2',
                title: 'tag2',
                is_active: 1
            }
        ],
        categories: [
            {
                category_id: 1,
                identifier: 'general',
                title: 'General',
                is_active: 1
            },
            {
                category_id: 2,
                identifier: 'new',
                title: 'New',
                is_active: 1
            }
        ]
    }
};

test('renders blog content', () => {
    const root = TestRenderer.create(
        <IntlProvider locale="en-US">
            <BlogContent {...blogContentProps} />
        </IntlProvider>
    );

    expect(root.toJSON()).toMatchSnapshot();
});
