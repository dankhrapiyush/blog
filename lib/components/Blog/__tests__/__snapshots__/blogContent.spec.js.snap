// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders blog content 1`] = `
Array [
  <h1
    className="title"
  >
    <div
      className="blogTitle"
    >
      Foo Bar
    </div>
  </h1>,
  <div
    className="postHolder"
  >
    <div
      className="postHeader"
    >
      <div
        className="postInfo"
      >
        <div
          className="item"
        >
          <span
            className="label"
          >
            <mock-FormattedMessage
              defaultMessage="Posted: "
              id="blogContent.posted"
            />
          </span>
          <span
            className="value"
          >
            Jul 10, 2020
          </span>
        </div>
        <div
          className="item"
        >
          <span
            className="label"
          >
            <mock-FormattedMessage
              defaultMessage="Categories: "
              id="blogsContent.categories"
            />
          </span>
          <span
            className="value"
          >
            <div
              to="/blog/category/general"
            >
              General
            </div>
            , 
            <div
              to="/blog/category/new"
            >
              New
            </div>
          </span>
        </div>
        <div
          className="item"
        >
          <span
            className="label"
          >
            <mock-FormattedMessage
              defaultMessage="Tags: "
              id="blogsContent.tags"
            />
          </span>
          <span
            className="value"
          >
            <div
              to="/blog/tag/tag1"
            >
              tag1
            </div>
            , 
            <div
              to="/blog/tag/tag2"
            >
              tag2
            </div>
          </span>
        </div>
        <div
          className="item"
        >
          <span
            className="label"
          >
            <mock-FormattedMessage
              defaultMessage="Author: "
              id="blogContent.author"
            />
          </span>
          <span
            className="value"
          >
            web dev
          </span>
        </div>
      </div>
    </div>
    <div
      className="postContent"
    >
      <div
        className="postTextHld"
      >
        <RichContent
          html="Foo Bar Description"
        />
      </div>
    </div>
  </div>,
]
`;
